packages/common/directives/src/filter/filter.directive.ts

```diff
@@ -1,37 +1,54 @@
-import { Directive, ElementRef, HostListener, Input, OnDestroy, OnInit } from '@angular/core';
-import { ControlValueInterceptor, ControlValueInterceptorDescriptor } from '@angular-ru/common/forms';
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+import { ControlValueInterceptor } from '@angular-ru/common/forms';
 import { filter, FilterPredicateFn } from '@angular-ru/common/string';
 
 @Directive({
     selector: '[filter]',
     providers: [ControlValueInterceptor]
 })
-export class FilterDirective implements OnInit, OnDestroy {
+export class FilterDirective {
     @Input() public filter: string[] | FilterPredicateFn | RegExp = [];
-    public preparedValue: string = '';
 
-    private controlValueOperator: ControlValueInterceptorDescriptor = {
-        toModelValue: (value: string): string => {
-            this.preparedValue = filter(value, this.filter);
-            return this.preparedValue;
+    private throttle: boolean = false;
```

Comment:
I think you need to name the variable differently, because the meaning is not very clear what it does and why

Reply:
agree, it is not a good way
changed on this
```
private isNotPreviousEvent(event: InputEvent): boolean {
    return this.imitative !== event;
}
```